src/coprocessor/memoset/mod.rs

```diff
@@ -0,0 +1,850 @@
+//! The `memoset` module implements a `MemoSet`.
```

Comment:
Nit: you've made the public API of this module quite sparse by making few members public, which is great.

It also means that if you were to annotate the top of this module with `#![deny(missing_docs)]` the amount of work you'd have to do to fix would be small.